knowledge-base/grid-custom-cell-formatting-with-radspreadprocessing.md

```diff
@@ -0,0 +1,144 @@
+* Select the desired cells - create a [CellSelection](https://docs.telerik.com/devtools/document-processing/libraries/radspreadprocessing/working-with-cells/accessing-cells-of-worksheet).
+
+* Add your desired cell modifications to the selected cells. The example below demonstrates adding cell fill, you can modify different [cell properties](https://docs.telerik.com/devtools/document-processing/libraries/radspreadprocessing/working-with-cells/get-set-clear-properties#cell-properties) based on the result you want to achieve.
+
+* [Export the modified `workbook` to a `MemoryStream`](https://docs.telerik.com/devtools/document-processing/knowledge-base/import-export-save-load-workbook#save-workbook-to-filestream-or-memorystream).
+
+* Pass that `MemoryStream` to the `args.Stream` of the `GridAfterExcelExportEventArgs`, so that the modifications can be saved to the actual exported file.
+
+````CSHTML
+@*Use RadSpreadProcessing to set background to table headers*@
+
+@using Telerik.Documents.Media
+@using Telerik.Documents.Common.Model
+@using Telerik.Windows.Documents.Spreadsheet.FormatProviders
+@using Telerik.Windows.Documents.Spreadsheet.FormatProviders.OpenXml.Xlsx
+@using Telerik.Windows.Documents.Spreadsheet.FormatProviders.TextBased.Csv
+@using Telerik.Windows.Documents.Spreadsheet.Model
+
+<TelerikGrid Data="@GridData" Pageable="true">
+    <GridToolBar>
+        <GridCommandButton Command="ExcelExport" Icon="file-excel">Export to Excel</GridCommandButton>
+        <label class="k-checkbox-label"><TelerikCheckBox @bind-Value="@ExportAllPages" />Export All Pages</label>
+    </GridToolBar>
+
+    <GridExport>
+        <GridExcelExport FileName="telerik-grid-export"
+                         AllPages="@ExportAllPages"
+                         OnAfterExport="@OnExcelAfterExport" />
+    </GridExport>
+
+    <GridColumns>
+        <GridColumn Field="@nameof(SampleData.ProductId)" Title="ID" Width="100px" />
+        <GridColumn Field="@nameof(SampleData.ProductName)" Title="Product Name" Width="300px" />
+        <GridColumn Field="@nameof(SampleData.UnitsInStock)" Title="In stock" Width="100px" />
+        <GridColumn Field="@nameof(SampleData.Price)" Title="Unit Price" Width="200px" />
+        <GridColumn Field="@nameof(SampleData.Discontinued)" Title="Discontinued" Width="100px" />
+        <GridColumn Field="@nameof(SampleData.FirstReleaseDate)" Title="Release Date" Width="300px" />
+    </GridColumns>
+</TelerikGrid>
+
+@code {
+    private async Task OnExcelAfterExport(GridAfterExcelExportEventArgs args)
+    {
+        //and and copy the bytes to a new MemoryStream
+        var bytes = args.Stream.ToArray();
+
+        var excelStream = new MemoryStream(bytes);
+
+        //create a format provider instance to call the import
+        XlsxFormatProvider formatProvider = new XlsxFormatProvider();
+
+        //import the stream to a workbook
+        Workbook workbook = formatProvider.Import(excelStream);
+
+        //create cell selection
```

Comment:
You can explain the code with a slightly different wording, for better understanding
```suggestion
        //select a range of cells
```